Add project() method to DependencyHandler for consistency with Dependencies (#36692)
gradle/gradle: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_9.adoc

}
----

==== Deprecation of `Dependencies.getProject()` method

The `getProject()` method on `Dependencies` has been deprecated and will be removed in Gradle 10.0.0.
`Dependencies` is used to configure dependencies in test suites, and `getProject()` was being used by internal Gradle code to retrieve a reference to the current project.

> This method was not intended for public use, but it's possible that it could be used unintentionally.

Reviewer: ```suggestion
While never intended for public use, it could be used unintentionally in build scripts.
```

For example, a configuration like the following may accidentally access this method:

[source,kotlin]
----
testing {
    suites {
        val integrationTest by registering(JvmTestSuite::class) {
            dependencies {
                implementation(project(project.path)) // `project.path` accesses `Dependencies.getProject()`
            }
        }
    }
}
----

> The configuration above may now fail to compile with an error in 9.5.0 if `-Werror` is set.

Reviewer: ```suggestion
NOTE: If you use the `-Werror` flag, this configuration will fail to compile in Gradle 9.5+.
```
